core/org.osate.xtext.aadl2/src/org/osate/xtext/aadl2/validation/Aadl2JavaValidator.java

```diff
@@ -3217,7 +3217,7 @@ private void checkExtensionAndRealizationHierarchy(ImplementationExtension imple
 		}
 		if (!isAncestor) {
 			error(implementationExtension, '\'' + typeOfParent.getQualifiedName() + "' is not an ancestor of '"
-					+ typeOfChild.getQualifiedName() + "'.");
+					+ typeOfChild != null ? typeOfChild.getQualifiedName() : "TYPE OF CHILD NOT FOUND" + "'.");
```

Comment:
A null check is already performed on `typeOfChild` a few lines earlier with a call to `Aadl2Util.isNull(typeOfChild)`. I think we should ignore this issue reported by SonarCloud.